model/map/src/main/java/org/keycloak/models/map/storage/ConcurrentHashMapStorageProvider.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage;
+
+import org.keycloak.Config.Scope;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.map.common.AbstractEntity;
+import org.keycloak.models.map.common.Serialization;
+import com.fasterxml.jackson.databind.JavaType;
+import java.io.File;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jboss.logging.Logger;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class ConcurrentHashMapStorageProvider implements MapStorageProvider {
+
+    private static class ConcurrentHashMapStorage<K, V> extends ConcurrentHashMap<K, V> implements MapStorage<K, V> {
+    }
+
+    private static final String PROVIDER_ID = "concurrenthashmap";
+
+    private static final Logger LOG = Logger.getLogger(ConcurrentHashMapStorageProvider.class);
+
+    private final ConcurrentHashMap<String, ConcurrentHashMap<?,?>> storages = new ConcurrentHashMap<>();
+
+    @Override
+    public MapStorageProvider create(KeycloakSession session) {
+        return this;
+    }
+
+    @Override
+    public void init(Scope config) {
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+    }
+
+    @Override
+    public void close() {
+        storages.forEach(ConcurrentHashMapStorageProvider::storeMap);
+    }
+
+    private static void storeMap(String fileName, ConcurrentHashMap<?, ?> store) {
+        if (fileName != null) {
+            final File f = new File("target/map-" + fileName + ".json");
```

Comment:
When running `KeycloakServer.java` from idea the working directory is set to keycloak root directory. Therefore this fails with `file or directory doesn't exists` because `target` directory is not there. Can we use `project.build.directory` variable? Also how this will work if we will run it on wildfly or using quarkus jar file? Should we provide some env variable to set this path?